packages/edit-post/src/components/sidebar/settings-header/index.js

```diff
@@ -1,65 +1,73 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { Button } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
-import { withDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+// translators: Label for the Block Settings Sidebar tab, not selected.
+const BLOCK_LABEL = __( 'Block' );
+// translators: Label for the Block Settings Sidebar tab, selected.
+const BLOCK_SELECTED_LABEL = __( 'Block (selected)' );
+
+const SettingsHeader = ( { sidebarName } ) => {
+	const { openGeneralSidebar } = useDispatch( 'core/edit-post' );
+
+	const documentLabel = useSelect( ( select ) => {
+		const currentPostType = select( 'core/editor' ).getCurrentPostType();
+		const postType = select( 'core' ).getPostType( currentPostType );
+
+		// translators: Default label for the Document sidebar tab, not selected.
+		const defaultDocumentLabel = __( 'Document' );
+
+		return get(
```

Comment:
Could use the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to avoid pulling the lodash dep:

```js
return postType?.labels?.singular_name ?? __( 'Document' );

Reply:
Unfortunately, I can't.

`postType?.labels?.singular_name` gives me eslint error. `Identifier 'singular_name' is not in camel case.eslintcamelcase`

The eslint error goes away if I use `postType.labels.singular_name` without the optional chaining. But then it defeats the purpose. Okay, if I keep it as is?

Reply:
That's really a false positive in this case. Just disable it :-)

```js
// Disable reason: Post type labels object is shaped like this.
// eslint-disable-next-line name-of-the-eslint-rule
return postType?.labels?.singular_name ?? __( 'Document' );
```